README.md

```diff
@@ -0,0 +1,72 @@
+# Scala LruMap
+
+[![Build Status](https://api.travis-ci.org/snowplow-incubator/lru-map.svg)](https://travis-ci.org/snowplow-incubator/lru-map)
+[![Maven Central](https://img.shields.io/maven-central/v/com.snowplowanalytics/lru-map_2.12.svg)](https://maven-badges.herokuapp.com/maven-central/com.snowplowanalytics/lru-map_2.12)
+[![codecov](https://codecov.io/gh/snowplow-incubator/lru-map/branch/master/graph/badge.svg)](https://codecov.io/gh/snowplow-incubator/lru-map)
+[![Join the chat at https://gitter.im/snowplow-incubator/lru-map](https://badges.gitter.im/snowplow-incubator/lru-map.svg)](https://gitter.im/snowplow-incubator/lru-map?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
+
+A pure least recently used hash map based on
+[`java.util.LinkedHashMap`][linkedhashmap].
+
+## Example Usage
+
+This library is useful if you're trying to memoize expensive methods.
+
+```scala
+import cats.effect.IO
+
+class Fibonacci(lru: LruMap[IO, BigInt, BigInt]) {
+  def calc(n: BigInt): IO[BigInt] = {
+    if (n <= 0)
+      IO.pure(0)
+    else if (n == 1)
+      IO.pure(1)
+    else
+      lru.get(n).flatMap(r => r match {
+        case Some(r) => IO.pure(r)
+        case None => for {
+          m1 <- calc(n-1)
+          m2 <- calc(n-2)
+          _  <- lru.put(n, m1+m2)
+        } yield m1+m2
+      })
+  }
+}
+
+val result = (for {
+  // When the size of the map exceeds 500 the least recently used element is
+  // removed
+  lru    <- LruMap.create[IO, BigInt, BigInt](500)
+  result <- (new Fibonacci(lru)).calc(100)
+} yield result).unsafeRunSync
```

Comment:
It is a very common convention to append `()` to `Unit`-side-effecting method invocation: `effect.unsafeRunSync()`